modules/node/handler/cms_node_pagination_builder.e

```diff
@@ -0,0 +1,44 @@
+note
+	description: "Paginator builder for CMS nodes."
+	date: "$Date$"
+	revision: "$Revision$"
+
+class
+	CMS_NODE_PAGINATION_BUILDER
+
+inherit
+
+	CMS_PAGINATION_BUILDER [CMS_NODE]
+
+create
+	make
+
+feature {NONE} -- Initialization
+
+	make (a_api: CMS_NODE_API)
+			-- Create an object.
+		do
+			node_api := a_api
+			count := 5
+			offset := 0
+		ensure
+			node_api_set: node_api = a_api
+			limit_set: count = 5
+			offset_set: offset = 0
+		end
+
+	node_api: CMS_NODE_API
+			-- CMS API.
+
+feature -- Pager
+
+	items: ITERABLE [CMS_NODE]
+			-- <Precursor>.
+		do
+				--NOTE: the current implementation does not use
```

Comment:
Just a though
what about mixing your solution, and a tiny part of my previous suggestion related to passing an optional object to the queries such as nodes, recent_nodes ... and that optional object could be similar to 
CMS_PAGINATION_BUILDER
but mostly containing the offset and count, order, and asc vs desc ..
and we can imagine in the future more options like ... include  trashed items and so on ...